svn di src/org/jruby/RubyThread.java src/org/jruby/Ruby.java
Index: src/org/jruby/RubyThread.java
===================================================================
--- src/org/jruby/RubyThread.java	(revision 4363)
+++ src/org/jruby/RubyThread.java	(working copy)
@@ -266,7 +266,7 @@
         }
     }
 
-    private RubyThread(Ruby runtime, RubyClass type) {
+    protected RubyThread(Ruby runtime, RubyClass type) {
         super(runtime, type);
         this.threadService = runtime.getThreadService();
         // set to default thread group
Index: src/org/jruby/Ruby.java
===================================================================
--- src/org/jruby/Ruby.java	(revision 4363)
+++ src/org/jruby/Ruby.java	(working copy)
@@ -67,6 +67,7 @@
 import org.jruby.compiler.NotCompilableException;
 import org.jruby.compiler.impl.StandardASMCompiler;
 import org.jruby.compiler.yarv.StandardYARVCompiler;
+import org.jruby.debug.RubyDebugBaseLibrary;
 import org.jruby.evaluator.EvaluationState;
 import org.jruby.exceptions.JumpException;
 import org.jruby.exceptions.RaiseException;
@@ -739,6 +740,7 @@
         
         registerBuiltin("socket.rb", new RubySocket.Service());
         registerBuiltin("rbconfig.rb", new RbConfigLibrary());
+        registerBuiltin("ruby_debug.so", new RubyDebugBaseLibrary());
 
         for (int i=0; i<BUILTIN_LIBRARIES.length; i++) {

             if(profile.allowBuiltin(BUILTIN_LIBRARIES[i])) {
